litellm/proxy/proxy_server.py

```diff
@@ -352,6 +352,24 @@ def _get_pydantic_json_dict(pydantic_obj: BaseModel) -> dict:
         return pydantic_obj.dict()
 
 
+def get_custom_headers(*,
+                       model_id: Optional[str] = None,
```

Comment:
please try/except this function. It should not block an LLM API call from returning

Reply:
@ishaan-jaff this does not block anything, just filters out the header with empty(`None`, `''`) values. Even if i add try-cache, it will never throw an exception based on the use case.

Reply:
If `get_custom_headers` raises an exception it will raise an exception on the LLM routes. You need to try/except this function

Reply:
This code can raise an error: 

```
{key: value for key, value in headers.items() if value not in exclude_values}
```

Reply:
gottcha, on it

Reply:
Updated it, thank you:

<img width="1244" alt="image" src="https://github.com/BerriAI/litellm/assets/4735661/e0b7a164-fb04-428b-bfb2-558082501d8a">
<img width="501" alt="image" src="https://github.com/BerriAI/litellm/assets/4735661/4c6c4790-6bfa-4745-9b4b-8aff57a853f7">